ui/app/controllers/jobs/index.js

```diff
@@ -53,8 +54,43 @@ export default class IndexController extends Controller.extend(
     },
   ];
 
-  currentPage = 1;
-  @readOnly('userSettings.pageSize') pageSize;
+  // START PAGINATION LOGIC
+  @tracked pageSize = this.userSettings.pageSize;
+  @tracked nextToken = null;
+  @tracked previousTokens = [];
+
+  get shouldDisableNext() {
+    return !this.model.jobs.meta?.nextToken;
+  }
+
+  get shouldDisablePrev() {
+    return !this.previousTokens.length;
+  }
+
+  @action
+  onNext(nextToken) {
+    this.previousTokens = [...this.previousTokens, this.nextToken];
+    this.nextToken = nextToken;
+  }
+
+  @action
+  onPrev(lastToken) {
+    this.previousTokens.pop();
+    this.previousTokens = [...this.previousTokens];
+    this.nextToken = lastToken;
+  }
+
+  @action
+  refresh() {
+    this.nextToken = null;
+    this.previousTokens = [];
+  }
```

Comment:
Clarification: I can't get it to work on the first job page. If I go to page 2 or further, it resets me back to page 1. This feels like it might be confusing behaviour to an end-user.

Reply:
I would expect that the button would make a re-request of my data (one might refresh because they suspect data has changed)